### ISSUE TITLE
Is `getGlobals` doing too much work?

### ISSUE THREAD

**Issue by arturmuller** [pavelvolek/redux-keen#2]
I have a feeling that relying on the `getGlobals` callback to set 'static' globals would not make a lot of sense to me if I came to this package for the first time.

What do you think about separating the static stuff (ip address, addons, etc...) to be added to the middleware on init (as a plain object), and in the callback **only deal with dynamic stuff**. Some people might not even need the dynamic properties, and the callback function is going to be much neater when concerns are separated like this.

The API could look like this:

``` js
import reduxKeen from 'redux-keen';

const projectId = '123123123213';
const writeKey = '123123123213';
const globals = { ip_address: '${keen.ip}' };
const loadDynamicProperties = (state) => {
  const { isSignedIn, user } = state.session;
  if (!isSignedIn) return;

  return {
    user: {
      first_name: `${user.firstName}`,
      last_name: `${user.lastName}`,
      user_id: user.profileId
    }
  };
};

const analyticsMiddleware = reduxKeen(projectId, writeKey, globals, loadDynamicProperties);

// ... and place it in the redux middleware stack.

```

**Comment by pavelvolek** [comment]
Yes, you're right and I like this! lets separate those two and try it with Oyster!

**Comment by arturmuller** [comment]
:+1:

**Comment by markdalgleish** [comment]
That's a lot of arguments for the middleware. You might want to move towards named arguments instead?

``` js
const analyticsMiddleware = reduxKeen(keenClient, { globals, loadDynamicProperties });
```

**Comment by arturmuller** [comment]
Yep, that's better! :+1:

Now that we have to actually name them, I am thinking what the names should be.
1. I think `globals` is actually not a great name. I think the key thing about these is that they are static, not that they are global. So what about `staticProperties`? Or maybe just be `properties`? I guess it is implied that they are static if it is just a plain object.
2. `loadDynamicProperties` is clear, but kind of a mouthful. So maybe `loadProperties`? `getProperties`?

This is my favourite so far:

``` js
const analyticsMiddleware = reduxKeen(keenClient, { properties, getProperties });
```

@pavelvolek?

**Comment by pavelvolek** [comment]
Well @arturmuller, I had the same thought. I played a bit with the implemantation already, and what I have is this: 

``` js
const analyticsMiddleware = reduxKeen(keenClient, { staticProperties, getDynamicProperties });
```

But I love your short version `static` and `dynamic` words omitted. As you say, properties are plain object thus static, and the getProperties is a function that implies it is something dynamic. So I would go with your naming: 

``` js
const analyticsMiddleware = reduxKeen(keenClient, { properties, getProperties });
```

**Comment by markdalgleish** [comment]
We're looking at an approach like this currently. We're talking about building an `analytics` reducer, then simply returning `state.analytics` inside of `getGlobals` (as it's currently known). Any "static" properties would exist in the initial state of the analytics reducer, like any other data in our app.

Personally, I'm not sure your API needs to do anything more than let me pluck some data from my Redux state tree, as you currently do. In fact, I think it might actually hurt maintenance if some of the event data is in `createStore`, as I view that file as plumbing code, not something that defines data like that.

Thoughts?

**Comment by arturmuller** [comment]
Ok. Gotcha. For us, the static properties include something like this:

```
{
    ip_address: '${keen.ip}',
    user_agent: '${keen.user_agent}',
    keen: {
      addons: [
        {
          name: 'keen:ip_to_geo',
          input: { ip: 'ip_address' },
          output: 'ip_geo_info',
        },
        {
          name: 'keen:ua_parser',
          input: { 'ua_string': 'user_agent' },
          output: 'parsed_user_agent',
        },
      ],
    },
  };
```

...which is essentially just telling Keen how to enrich our stuff. In my eyes, this actually kinda is 'plumbing', it just so happens that Keen needs it attached to every event.

What do you think? For me, having the above in the global application state feels odd, but maybe I am missing something?
